Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -63,52 +64,52 @@ class VirtualMachineSlot: NSObject, ObservableObject, VZVirtualMachineDelegate {
         } catch {
             Logger.helper.error("Error launching VM: \(error.localizedDescription)")
             Logger.helper.error("Attempting Cleanup of \(launchConfiguration.handle)")
-            try await vmManager.removeVM(name: launchConfiguration.handle)
 
+            try? await VMManager.removeVM(name: launchConfiguration.handle)
             self.status = .crashed(error)
             throw error
         }
     }
 
     @MainActor
     func stopVirtualMachine() async throws {
-        self.status = .stopping
-        try await virtualMachine?.stop()
-        self.status = .empty
+        switch status {
+        case .starting, .running:
+            self.status = .stopping
+        default:
+            break
+        }
+        try await virtualMachine?.instance.stop()
+        await resetSlot()
     }
 
-    /// Handle a VM that was stopped from outside this object (for instance – by being shut down internally)
+    /// Resets the slot to a stopped status
     ///
     /// No need to do anything except some internal bookkeeping
-    ///
+    /// - Parameter error: Error supplied if the VM crashed
     @MainActor
-    func flush() async throws {
-        self.virtualMachine = nil
-        self.status = .empty
-    }
+    func resetSlot(withError error: Error? = nil) async {
```

Comment:
Note no `throws` so it doesn't get hung up on tries to get back to the original state.